src/OVAL/probes/CMakeLists.txt

```diff
@@ -0,0 +1,241 @@
+    target_link_libraries(probe_iflisteners probe)
+    install(TARGETS probe_iflisteners DESTINATION ${PROBE_INSTALL_DIR})
+
+    if(SELINUX_FOUND)
+	add_executable(probe_selinuxboolean "unix/linux/selinuxboolean.c")
+	target_include_directories(probe_selinuxboolean PRIVATE ${SELINUX_INCLUDE_DIR})
+	target_link_libraries(probe_selinuxboolean probe ${SELINUX_LIBRARIES})
+	install(TARGETS probe_selinuxboolean DESTINATION ${PROBE_INSTALL_DIR})
+
+	add_executable(probe_selinuxsecuritycontext "unix/linux/selinuxsecuritycontext.c")
+	target_include_directories(probe_selinuxsecuritycontext PRIVATE ${SELINUX_INCLUDE_DIR})
+	target_link_libraries(probe_selinuxsecuritycontext probe ${SELINUX_LIBRARIES})
+	install(TARGETS probe_selinuxsecuritycontext DESTINATION ${PROBE_INSTALL_DIR})
+    endif()
+
+	if(RPM_FOUND)
+		add_executable(probe_rpminfo "unix/linux/rpminfo.c" "unix/linux/rpm-helper.h" "unix/linux/rpm-helper.c")
+		target_link_libraries(probe_rpminfo probe)
+		install(TARGETS probe_rpminfo DESTINATION ${PROBE_INSTALL_DIR})
+
+		add_executable(probe_rpmverify "unix/linux/rpmverify.c" "unix/linux/rpm-helper.h" "unix/linux/rpm-helper.c")
+		target_link_libraries(probe_rpmverify probe)
+		install(TARGETS probe_rpmverify DESTINATION ${PROBE_INSTALL_DIR})
+
+		add_executable(probe_rpmverifyfile "unix/linux/rpmverifyfile.c" "unix/linux/rpm-helper.h" "unix/linux/rpm-helper.c")
+		target_link_libraries(probe_rpmverifyfile probe)
+		install(TARGETS probe_rpmverifyfile DESTINATION ${PROBE_INSTALL_DIR})
+
+		add_executable(probe_rpmverifypackage "unix/linux/rpmverifypackage.c" "unix/linux/rpm-helper.h" "unix/linux/rpm-helper.c" "unix/linux/probe-chroot.h" "unix/linux/probe-chroot.c")
+		target_link_libraries(probe_rpmverifypackage probe ${POPT_LIBRARIES})
+		install(TARGETS probe_rpmverifypackage DESTINATION ${PROBE_INSTALL_DIR})
+	endif()
+
+    if (APTPKG_FOUND)
+	add_executable(probe_dpkginfo "unix/linux/dpkginfo.c" "unix/linux/dpkginfo-helper.cxx" "unix/linux/dpkginfo-helper.h")
+	target_include_directories(probe_dpkginfo PRIVATE ${APTPKG_INCLUDE_DIR})
+	target_link_libraries(probe_dpkginfo probe ${APTPKG_LIBRARIES})
+	install(TARGETS probe_dpkginfo DESTINATION ${PROBE_INSTALL_DIR})
+    endif()
+
+    if(DBUS_FOUND)
+	add_executable(probe_systemdunitproperty "unix/linux/systemdunitproperty.c" "unix/linux/systemdshared.h")
+	target_include_directories(probe_systemdunitproperty PRIVATE ${DBUS_INCLUDE_DIRS})
+	target_link_libraries(probe_systemdunitproperty probe ${DBUS_LIBRARIES})
+	install(TARGETS probe_systemdunitproperty DESTINATION ${PROBE_INSTALL_DIR})
+
+	add_executable(probe_systemdunitdependency "unix/linux/systemdunitdependency.c" "unix/linux/systemdshared.h")
+	target_include_directories(probe_systemdunitdependency PRIVATE ${DBUS_INCLUDE_DIRS})
+	target_link_libraries(probe_systemdunitdependency probe ${DBUS_LIBRARIES})
+	install(TARGETS probe_systemdunitdependency DESTINATION ${PROBE_INSTALL_DIR})
+    endif()
+endif()
```

Comment:
I am not a big fan of how this is implemented. The old ways of switching things on and off are definitely not ideal, you can choose something better. Definitely think about making a macro or a function. This file is 99% copy paste repeated patterns.

Do users need to turn of all unix probes at once? Isn't it better to create a cmake option for each and every probe and let users select whichever subset of probes they want? I do not have answers to these questions, I think most users just run configure and keep the default settings.